### PR TITLE
drivers: Fix missing `auto_attach_cmd`

### PR DESCRIPTION
**Pull request by LingaoM**
When run native posix with `-attach_uart_cmd=<cmd>`,
should attach cmd by arguments.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

**Review by aescolar — DISMISSED** [review]
It makes sense to do this, but we should say in:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_native_posix.c#L402
That it "(Implies attach_uart)"

If it were me, instead of ORing the option here, I would add a callback to the `auto_attach_cmd` option which sets `auto_attach`. (it would be a bit more code, but cleaner for whoever reads it later)

**Comment by LingaoM** [comment]
> It makes sense to do this, but we should say in:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_native_posix.c#L402
> 
> That it "(Implies attach_uart)"
> 
> 
> 
> If it were me, instead of ORing the option here, I would add a callback to the `auto_attach_cmd` option which sets `auto_attach`. (it would be a bit more code, but cleaner for whoever reads it later)

auto_attach and auto_attach_cmd are different in kconfig file, also use different arguments when run.

Therefore, I think it is a tie relationship rather than an inheritance relationship

So, this is my original intention.

**Comment by LingaoM** [comment]
> It makes sense to do this, but we should say in:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_native_posix.c#L402
> 
> That it "(Implies attach_uart)"
> 
> 
> 
> If it were me, instead of ORing the option here, I would add a callback to the `auto_attach_cmd` option which sets `auto_attach`. (it would be a bit more code, but cleaner for whoever reads it later)

In fact, your approach makes this inheritance relationship more clear.  It should be the same, but it shouldn't just be implied, it should be clear.

File: drivers/serial/uart_native_posix.c [hunk 1]
```diff
@@ -398,7 +403,7 @@ static void np_add_uart_options(void)
 		"Automatically attach to the UART terminal"},
 		{false, false, false,
 		"attach_uart_cmd", "\"cmd\"", 's',
-		(void *)&auto_attach_cmd, NULL,
+		(void *)&auto_attach_cmd, auto_attach_cmd_cb,
 		"Command used to automatically attach to the terminal, by "
```
Comment (aescolar) [inline drivers/serial/uart_native_posix.c:407]: Would you be so kind to add a "(Implies auto_attach)" to the this help description, so it is clearer for users.

File: drivers/serial/uart_native_posix.c [hunk 2]
```diff
@@ -395,12 +400,13 @@ static void np_add_uart_options(void)
 		{false, false, true,
 		"attach_uart", "", 'b',
 		(void *)&auto_attach, NULL,
-		"Automatically attach to the UART terminal"},
+		"Automatically attach to the UART terminal, by "
+		"default: '" CONFIG_NATIVE_UART_AUTOATTACH_DEFAULT_CMD "'"},
```
Comment (aescolar) [inline drivers/serial/uart_native_posix.c:404]: why did you move this line?

Comment (LingaoM) [reply drivers/serial/uart_native_posix.c:404]: In my opinion, if only this parameter is specified, what is the default command to be executed.

Comment (aescolar) [reply drivers/serial/uart_native_posix.c:404]: The default command is what you can override with auto_attach_cmd. The help message is printed in the same order as listed here. So to me it seems clearer as it was originally.
(If you leave this line where it was I can approve this PR :) )

File: drivers/serial/uart_native_posix.c [hunk 3]
```diff
@@ -378,6 +378,11 @@ DEVICE_DT_INST_DEFINE(1,
 	    &np_uart_driver_api_1);
 #endif /* CONFIG_UART_NATIVE_POSIX_PORT_1_ENABLE */
 
+static void auto_attach_cmd_cb(char *argv, int offset)
+{
```
Comment (aescolar) [inline drivers/serial/uart_native_posix.c:382]: Oh, I missed this: Please add
```C
	ARG_UNUSED(argv);
	ARG_UNUSED(offset);
```

**Review by aescolar — APPROVED** [review]

**Review by daor-oti — APPROVED** [review]
